src/transform/src/nonnullable.rs

```diff
@@ -28,30 +29,38 @@ impl crate::Transform for NonNullable {
         &self,
         relation: &mut MirRelationExpr,
         _: TransformArgs,
-    ) -> Result<(), crate::TransformError> {
+    ) -> Result<(), TransformError> {
         relation.try_visit_mut_pre(&mut |e| self.action(e))
     }
 }
 
 impl NonNullable {
     /// Harvests information about non-nullability of columns from sources.
-    pub fn action(&self, relation: &mut MirRelationExpr) -> Result<(), crate::TransformError> {
+    pub fn action(&self, relation: &mut MirRelationExpr) -> Result<(), TransformError> {
         match relation {
             MirRelationExpr::Map { input, scalars } => {
-                if scalars.iter().any(scalar_contains_isnull) {
+                let contains_isnull = scalars
+                    .iter()
+                    .map(scalar_contains_isnull)
+                    .fold_ok(false, |b1, b2| b1 || b2)?;
```

Comment:
Well, what about a loop? :D

```rust
let mut contains_isnull = false;
for scalar in &scalars {
    if scalar_contains_isnull(scalar)? {
        contains_isnull = true;
        break;
    }
}
```

It's probably possible to do this with iterators somehow, but I doubt it would be more readable.

Reply:
Indeed. This will change the style in this from functional to imperative, though. @aalexandrov Would that be OK? I can change it in the three occurrences in nonnullable.rs if you find the style change does not hurt readability / maintainability.